src/Query/Query.php

```diff
@@ -251,21 +251,19 @@ public function getMessage($msgno, $msglist = null){
      * @throws GetMessagesFailedException
      */
     public function paginate($per_page = 5, $page = null, $page_name = 'imap_page'){
-        $this->page = $page > $this->page ? $page : $this->page;
-        $this->limit = $per_page;
-
-        $messages = $this->get();
-        if (($count = $messages->count()) > 0) {
-            $limit = $this->limit > $count ? $count : $this->limit;
-            $collection = array_fill(0, $messages->total() - $limit, true);
-            $messages->each(function($message) use(&$collection){
-                $collection[] = $message;
-            });
-        }else{
-            $collection = array_fill(0, $messages->total(), true);
+        if (
+               $page === null
+            && isset($_GET[$page_name])
+            && $_GET[$page_name] > 0
```

Comment:
I still think this condition is necessary (instead of ternary condition) because if `$_GET[$page_name]` isn't positive, `Query::$page` shouldn't be touched and should remain to it's default value: 1

Reply:
Wouldn't `$_GET[$page_name]` throw a notice  if it isn't set?

Reply:
If `$_GET[$page_name]` isn't set, a notice should be raised.

L:256 checks if `$_GET[$page_name]` is set. If not, L:257 won't be checked, so it would jump the block and try the next condition (hence, without touching `$this->page` which would remain to `1`)